core/src/main/java/org/testng/internal/FactoryMethod.java

```diff
@@ -181,22 +181,23 @@ private void init(Object instance, IAnnotationFinder annotationFinder, Construct
                     com.getDeclaringClass().getName(), com.getName());
           }
           if (indices == null || indices.isEmpty()) {
+            int finalPosition = position;
             result.addAll(Arrays.stream(testInstances).map(instance ->
-              new ParameterInfo(instance, parameters)
+              new ParameterInfo(instance, finalPosition, parameters)
```

Comment:
Why not directly using `position` here?

Reply:
![image](https://user-images.githubusercontent.com/60096114/114394024-65832480-9ba3-11eb-98ad-7ffe08d4e660.png)